mbrl/models/model_trainer.py

```diff
@@ -117,14 +124,30 @@ def train(
         epochs_since_update = 0
         best_val_score = self.evaluate(eval_dataset)
         for epoch in epoch_iter:
+            if batch_callback:
+                batch_callback_epoch = functools.partial(batch_callback, epoch)
+            else:
+                batch_callback_epoch = None
             batch_losses: List[float] = []
             for batch in dataset_train:
-                loss = self.model.update(batch, self.optimizer)
+                loss_and_maybe_meta = self.model.update(batch, self.optimizer)
+                if isinstance(loss_and_maybe_meta, tuple):
+                    loss = cast(float, loss_and_maybe_meta[0])
+                    meta = cast(Dict, loss_and_maybe_meta[1])
+                else:
+                    # TODO remove this if in v0.2.0
+                    warnings.warn(_NO_META_WARNING_MSG)
+                    loss = cast(float, loss_and_maybe_meta)
+                    meta = None
                 batch_losses.append(loss)
+                if batch_callback_epoch:
+                    batch_callback_epoch(loss, meta)
```

Comment:
I think it might make sense to pass the number of the current batch to the callback as well. E.g. if the epochs are really long, a user might want to log every x batches or just print some kind of progress bar with the number of the current batch. Of course one can keep some kind of counter inside the callback to count the batches, but that would be a bit ugly.
The same changes would also be necessary in `ModelTrainer.evaluate()`.

Reply:
I think there is a balance, since having a lol of arguments also makes the API more rigid. My inclination is to err on the side of flexibility, even if it requires the user to do slightly more work. The epoch counter I think is necessary, since otherwise there is no way for the user to check which epoch it is just by looking at what the callback gets.

Reply:
I think the batch index as additional argument would not be too critical. But I see your point. Especially since the batch index is not really needed in a typical use case (per episode logging should be sufficient) and there is the workaround with the counter for the special cases. So I am fine with not adding this argument.